packages/core/src/lib/implementation/read-rc-file.integration.test.ts

```diff
@@ -0,0 +1,32 @@
+import { dirname, join } from 'path';
+import { fileURLToPath } from 'url';
+import { describe, expect } from 'vitest';
+import { readRcByPath } from './read-rc-file';
+
+describe('readRcByPath', () => {
+  const configDirPath = join(
+    fileURLToPath(dirname(import.meta.url)),
+    '..',
+    '..',
+    '..',
+    '..',
+    '..',
+    'testing-utils',
+    'src',
+    'lib',
+    'fixtures',
+    'configs',
+  );
+
+  it('should throw if the configuration is empty', async () => {
+    await expect(
+      readRcByPath(join(configDirPath, 'code-pushup.empty.config.js')),
+    ).rejects.toThrow(`"code": "invalid_type",`);
+  });
+
+  it('should throw if the configuration is invalid', async () => {
+    await expect(
+      readRcByPath(join(configDirPath, 'code-pushup.invalid.config.ts')),
+    ).rejects.toThrow(/refs are duplicates/);
+  });
+});
```

Comment:
It's weird not to have a happy path test case.

Reply:
yes reason was mocking is hard atm as it is always everywhere... and in this case filesystem as well as bundle require is used

Reply:
it is in now